server/ports.go

```diff
@@ -0,0 +1,81 @@
+package server
+
+import (
+	"fmt"
+	"sync"
+)
+
+// portManager is responsible for maintaining segregated access to the port field
+// via the use of rwLock sync.RWMutex and mustRead sync.Mutex
+// rwLock establishes a standard read write mutex that allows consecutive reads and exclusive writes
+// mustRead forces MustGetPort to wait until port has a none 0 value
+type portManger struct {
+	port             int
+	afterPortChanged func(port int)
+	rwLock           *sync.RWMutex
+	mustRead         *sync.Mutex
+}
+
+// newPortManager returns a newly initialised portManager with a pre-Locked portManger.mustRead sync.Mutex
+func newPortManager(afterPortChanged func(int)) portManger {
+	pm := portManger{
+		afterPortChanged: afterPortChanged,
+		rwLock:           new(sync.RWMutex),
+		mustRead:         new(sync.Mutex),
```

Comment:
Out of curiosity: doing `new(SomeType)` is just another way of doing `&SomeType{}` ?

Reply:
Yes, that's correct. It is kind of the "correct" way to do it, but it is generally fine to declare a struct literal pointer. The main advantage of declaring a with `new` is that it can work for any literal pointer type, see this example of its usefulness https://go.dev/play/p/5OoeMXnJsb-